src/ethernet_interface.cpp

```diff
@@ -1056,14 +1056,19 @@ void EthernetInterface::writeConfigurationFile()
     // with eth0 route.
     // https://github.com/systemd/systemd/issues/23595
     // TODO: Remove this check once 23595 is resolved
-    if (interfaceName() != "eth1")
-    {
 #ifdef LINK_LOCAL_AUTOCONFIGURATION
+    if (interfaceName() == "eth0")
```

Comment:
is there any reason for link local addressig has different values for eth0 and eth1?